### DISCUSSION TITLE
Runtime style compilation

### DISCUSSION THREAD

**Discussion by ianwijma** [tailwindlabs/tailwindcss#4684] [Ideas]
This is probably vastly out of scope of tailwindCSS.

## This issue

I have dynamic background images, where I want to use react to set the background image in the following way. Which ATM obviously doesn't work.

```jsx
return <div className={`bg-cover bg-cover bg-no-repeat bg-image-[${src}]`} />
```

But this won't work. So I need to resort to setting it on the style attribute.

```jsx
return <div className='bg-cover bg-cover bg-no-repeat' style={{ backgroundImage: `url('${url}')` }} />
```

## The idea

The idea is to have some client sided code which track down these elements and create correct styles on the fly. 

Again, this is probably vastly out of scope. but I'd just think it would be such a cool addition.

**Comment by VicGUTT** [comment]
Hey !

- Related issue : https://github.com/tailwindlabs/tailwindcss/issues/3941
- First PR attempt at something similar : https://github.com/tailwindlabs/tailwindcss-jit/pull/176

Let's keep that idea growing ! 💪🏾

**Reply by ianwijma** [reply]
OW nice! Glad other people have related idea! Loving it!

**Comment by ianwijma** [comment]
My use case is with NextJS, SSG. So it's not fully dynamic. Only during build time.
